dotnet/src/SemanticKernel/Planning/IPlan.cs

```diff
@@ -0,0 +1,85 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Collections.Generic;
+using System.Text.Json.Serialization;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.SemanticKernel.Orchestration;
+
+namespace Microsoft.SemanticKernel.Planning;
+
+/// <summary>
+/// Interface for a plan executable by the kernel
+/// </summary>
+public interface IPlan
+{
+    /// <summary>
+    /// State of the plan
+    /// </summary>
+    ContextVariables State { get; }
+
+    /// <summary>
+    /// Root step of the plan
+    /// </summary>
+    PlanStep Root { get; }
+
+    /// <summary>
+    /// Run the next step of the plan
+    /// </summary>
+    /// <param name="kernel">Kernel instance to use</param>
+    /// <param name="variables">Variables to use</param>
+    /// <param name="cancellationToken">Cancellation token</param>
+    /// <returns>The updated plan</returns>
+    Task<IPlan> RunNextStepAsync(
+        IKernel kernel,
+        ContextVariables variables,
+        CancellationToken cancellationToken = default);
+}
+
+/// <summary>
+/// An executable step in a plan
+/// </summary>
+public class PlanStep
+{
+    /// <summary>
+    /// Description of the step
+    /// </summary>
+    [JsonPropertyName("description")]
+    public string Description { get; set; } = string.Empty;
```

Comment:
What are the constraints on this and the other properties e.g., maximum length, is required, ...